geode-core/src/main/java/org/apache/geode/internal/cache/wan/AbstractGatewaySenderEventProcessor.java

```diff
@@ -146,6 +146,10 @@
    */
   private int batchSize;
 
+  private String expectedReceiverUniqueId = "";
```

Comment:
I think the expectedReceiverUniqueId should be on AbstractGatewaySender (like serverLocation).